### PR TITLE
Punish students for saying banned words

### PR DESCRIPTION
**Pull request by braxtonhall**
> **TA**:  !ban anime
> **bot**: Banned `anime`
> **student**: Hey all wanna watch anime?
> **bot**: @ student you should know better. Think about what you've done

Student then has message permissions taken away for 5 minutes

### CODE REVIEW

File: src/listeners/punishBan.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+const FIVE_MINUTES_MS = 5 * 60 * 1000;
+
+const punish: Listener<"messageCreate"> = {
+    event: "messageCreate",
+    procedure: async (client: Client, message: Message) => {
+        const bannedWord = await getBannedWord();
+        if (!!bannedWord && message.content.toLowerCase().includes(bannedWord)) {
+            if (await isCourseStaff(message)) {
+                Log.debug("Course staff said a bad word but we're letting it go bc power trip time");
```
Comment (falkirks) [inline src/listeners/punishBan.ts:14]: This doesn’t seem fair :P

File: src/listeners/punishBan.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            } else if  (message.author.bot) {
+                Log.debug("Bot said a bad word but we're letting it go bc it doesn't know any better");
+            } else {
+                const member = await message.guild.members.fetch(message.author.id);
+                // The smart thing to do here would be journal this in the database but let's just take a chance
+                const oldRoles = Array.from(member.roles.valueOf().keys())
+                    .filter((role) => member.guild.roles.everyone.id !== role);
+                await handlePunishment(member, oldRoles);
+                // Pray there is no server restart at this time lol
```
Comment (falkirks) [inline src/listeners/punishBan.ts:23]: I love this

**Review by falkirks — APPROVED** [review]

File: src/commands/bans/ban.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import {Command} from "@ubccpsc310/bot-base";
+import {Client, Message} from "discord.js";
+import {ban as banImpl} from "../../controllers/BanController";
+import {ConfigKey, getConfig} from "../../util/Config";
+
+const ban: Command = {
+    name: "ban",
+    description: "Bans a word",
+    usage: "ban <word>",
+    procedure: async (client: Client, message: Message, args: string[]) => {
```
Comment (jyoo980) [inline src/commands/bans/ban.ts:10]: Not sure which version of TS/ECMA you're on, but is top-level `await` permitted now?

Comment (braxtonhall) [reply src/commands/bans/ban.ts:10]: james. that says `async`.

Comment (jyoo980) [reply src/commands/bans/ban.ts:10]: Did I stutter?

**Pull request merged**